src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import { postComment } from "./Api";
+import "./Comments.css";
+const CommentAdder = ({ setComments, article_id }) => {
+  const [newCommentText, setNewCommentText] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    const newComment = {
+      body: newCommentText,
+      article_id,
+    };
+    postComment(newComment).then((newComment) => {
+      setComments((currComments) => {
+        setNewCommentText('')
+        setIsLoading(false)
+        return [newComment, ...currComments];
+      });
+    }).catch((err) => {
+      return <p>oops, something went wrong</p>
```

Comment:
This will never render unfortunately. Please use an error state and make sure you deal with this after the final return. Here you are only returning from the callback function, not from the Component.

Reply:
thank you I've refactored it .